libevmasm/Inliner.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+/**
+ * @file Inliner.cpp
+ * Inlines small code snippets by replacing JUMP with a copy of the code jumped to.
+ */
+
+#include <libevmasm/Inliner.h>
+
+#include <libevmasm/AssemblyItem.h>
+#include <libevmasm/SemanticInformation.h>
+
+#include <libsolutil/CommonData.h>
+
+#include <range/v3/view/drop.hpp>
+#include <range/v3/view/enumerate.hpp>
+#include <range/v3/view/slice.hpp>
+
+#include <libsolutil/Common.h>
+
+#include <cstddef>
+#include <map>
+#include <optional>
+
+
+using namespace std;
+using namespace solidity;
+using namespace solidity::evmasm;
+
+void Inliner::optimise()
+{
+	std::map<u256, AssemblyItems> inlinableBlocks;
```

Comment:
Maybe use a vector view or something? The original m_items lives until the end.

Reply:
Yep - originally I wanted to modify the original one in place, but then noticed that's rather insane, but now views should indeed do!

Reply:
Ah, ok... then I have to copy items into ``newItems`` in the end instead of moving them, though... but copying ``AssemblyItem`` shouldn't be much more expensive than moving, so still better.